### ISSUE TITLE
Question about intrinsics gradient flow and view indexing in erayzer.py

### ISSUE THREAD

**Issue by ranrhuang** [QitaoZhao/E-RayZer#12]
Hi, thanks for this excellent work! I have a couple of questions regarding the implementation in erayzer.py.

1. Intrinsics detachment and gradient flow

In Line 334:
`pred_fxfycxcy = rearrange(pred_fxfycxcy, '(b v) d -> b v d', b=b).detach()`

pred_fxfycxcy is detached from the computation graph. My question is: what is the motivation for detaching it here? If the intrinsics prediction is detached, it seems that gradients cannot propagate back to the intrinsics prediction branch. Does this mean the intrinsics are not optimized through the reconstruction loss? Or is there another pathway through which gradients are expected to flow?

2. View indexing (v vs v_all)

From Lines 338–351, I was wondering about the definition of v. I expect it to be v_input, but it seems to be equal to v_all.

Specifically:
```
c2w_input = pred_c2w[:, :v, ...]
c2w_target = pred_c2w[:, :v]
```
could you clarify why both c2w_input and c2w_target use the same slicing :v?

Thanks again for sharing this work! Any clarification would be greatly appreciated.

**Comment by QitaoZhao** [comment]
Hi, thanks for the great catches!

1. I believe this was an unintended mistake made by the coding agent when I refactored the inference code from our internal codebase. While this behavior does not affect inference, I am pretty sure that during training we removed the detach, so the reconstruction loss can back-propagate and optimize the intrinsics.
2. This part is only used for inference visualization, to make the rendered video appear continuous between the input views. At inference time, we usually do not have target-view cameras, so the input cameras are used as targets instead. During training, however, we render from the target cameras and apply supervision there, as described in the paper.

**Comment by ranrhuang** [comment]
Thanks for your reply!
